### PR TITLE
Clean up startup and tests

### PR DESCRIPTION
**Pull request by imurchie**
Much of this is just getting the tests running correctly on Travis.

Substantive changes:
1. Adds `uiautomator2ServerInstallTimeout` capability, to configure the amount of time to wait for the instrumentation package to be available on the device
2. Check that the instrumentation package is on the device. On local systems this happened almost immediately. On slower ones (like Travis) it took a while, and so we were already trying to launch it and the launching failed with no instrumentation package available.

### CODE REVIEW

File: lib/uiautomator2.js [hunk 1]
```diff
@@ -26,36 +28,80 @@ class UiAutomator2Server {
...
+
+    logger.debug(`Waiting up to ${retries * 1000}ms for instrumentation '${INSTRUMENTATION_TARGET}' to be available`);
+    let output;
+    try {
+      await retryInterval(retries, 1000, async () => {
+        try {
+          output = await this.adb.shell(['pm', 'list', 'instrumentation']);
+          if (output.indexOf('Could not access the Package Manager') !== -1) {
+            output = null; // remove output, so it is not printed below
+            throw new Error(`Problem running package manager: ${output}`);
```
Comment (jlipps) [inline lib/uiautomator2.js]: the message will have to be constructed before output is set to `null`, no?

File: lib/uiautomator2.js [hunk 2]
```diff
@@ -26,36 +28,80 @@ class UiAutomator2Server {
...
+          } else if (output.indexOf(INSTRUMENTATION_TARGET) === -1) {
+            throw new Error('No instrumentation process found. Retrying...');
+          }
+          logger.debug(`Instrumentation '${INSTRUMENTATION_TARGET}' available`);
+        } catch (err) {
```
Comment (jlipps) [inline lib/uiautomator2.js]: the fact that we do nothing with `err` other than throw it makes me think we don't need this to be in a try/catch block.

File: lib/uiautomator2.js (hunk 1 above)
Comment (imurchie) [inline lib/uiautomator2.js]: Yup.

File: lib/uiautomator2.js (hunk 2 above)
Comment (imurchie) [inline lib/uiautomator2.js]: Ah, yes. This was a holdover from the long process of debugging this stuff.

**Pull request merged**